src/commands/Minion/alch.ts

```diff
@@ -26,11 +26,94 @@ const unlimitedFireRuneProviders = resolveItems([
 	'Tome of fire'
 ]);
 
+export function calculateAlchItemByTripDuration(params: {
+	user: KlasaUser;
+	duration: number;
+	items?: { item: Item; qty: number }[];
+	qtyLimit?: number;
+	calculateByRunesOwned?: boolean;
+}) {
+	let { items, duration, user, qtyLimit, calculateByRunesOwned } = params;
+	const userBank = user.bank().clone();
+
+	if (!items) items = [];
+
+	if (items.length === 0) {
+		for (const i of user.getUserFavAlchs() as Item[]) {
+			// Ignore alchs that are already on the alch items
+			if (items.find(a => a.item.id === i.id)) continue;
+			items.push({
+				item: i,
+				qty: qtyLimit
+					? qtyLimit < userBank.amount(i.id)
+						? qtyLimit
+						: userBank.amount(i.id)
+					: userBank.amount(i.id)
+			});
+		}
+		if (items.length === 0) return false;
```

Comment:
This would have to be moved outside the if() block and changed to count the allItemsToAlch() array instead.